testing/tests/DevExpress.ui.widgets.dataGrid/filterPanel.tests.js

```diff
@@ -309,4 +331,44 @@ QUnit.module("Filter Panel", {
         // assert
         assert.deepEqual(result, "Not ([Field] Equals '1' And [Field] Equals '2')");
     });
+
+    QUnit.test("filterBuilder customOperation", function(assert) {
+        // arrange
+        var filter = ["dateField", "testOperation"],
+            customFilter = ["dateField", "=", "10/10/2010"],
+            customExpressionCounter = 0;
+
+        // act, assert
+        this.initFilterPanelView({
+            filterPanel: {
+                visible: true,
+            },
+            dataSource: [],
+            filterValue: filter,
+            filterBuilder: {
+                customOperations: [{
+                    name: "testOperation",
+                    caption: "TestOperation",
+                    dataTypes: ["date"],
+                    icon: "check",
+                    hasValue: false,
+                    calculateFilterExpression: function(operation, obj) {
+                        ++customExpressionCounter;
+
+                        // assert
+                        assert.equal(obj.caption, "Date");
+                        return customFilter;
+                    }
+                }]
+            },
+            columns: [
+                { dataField: "field", caption: "Field" },
+                { dataField: "dateField", caption: "Date", dataType: "date" },
+            ]
+        });
+
+        // assert
+        assert.equal(this.filterPanelView._getTextElement().text(), "[Date] TestOperation", "filterPanel text");
```

Comment:
I think element().find("." + FILTER_PANEL_TEXT_CLASS) is better to use than calling of private method _getTextElement()

Reply:
Good idea. I'll correct this.